### ISSUE TITLE
Support for Set and Map in model classes

### ISSUE THREAD

**Issue by cmelchior** [realm/realm-java#759]
Some model classes uses Set and Map. We should look into providing both options:

```
public class ExampleModel extends RealmObject {
  RealmSet<anyRealmDataType> realmSet;
  RealmMap<anyRealmDataType, anyRealmDataType> realmMap;
}
```

**Comment by cmelchior** [comment]
Also worth looking into people using their own collection types: https://twitter.com/tsmith/status/689568188514762754

Supporting the generic Collection interface / types would probably not be possible at all since we would have no way of redirecting access to the native code, but creating `AbstractSet`, `AbstractMap` and the likes might be possible although it uncertain if the effort is worth it.

**Comment by bmunkholm** [comment]
For those looking for a workaround until this is solves, see here: http://stackoverflow.com/questions/35311755/realm-orm-how-to-deal-with-maps

**Comment by marioguerriero** [comment]
+1 Set/Map support is crucial in a modern DBMS

**Comment by biagionicp** [comment]
+1

**Comment by mhmtk** [comment]
+1

**Comment by ishaforenko** [comment]
+1

**Comment by mikeyxkcd** [comment]
+1

**Comment by agrusha** [comment]
+1

**Comment by Zhuinden** [comment]
You guys really *do* need `RealmSet` now.

**Comment by Zhuinden** [comment]
Hello, any chance of a `RealmLinkedSet`?

**Comment by cmelchior** [comment]
@Zhuinden Not impossible, I assume it is because you are looking for insertion order iteration?

**Comment by Zhuinden** [comment]
@cmelchior yes, but without having to worry about duplication on `add()`.

**Comment by asiljan** [comment]
+1

**Comment by valentin-sobolev** [comment]
+1

**Comment by Bastosss77** [comment]
+10000000

**Comment by jennymolske** [comment]
+1 👍

**Comment by lauraoran** [comment]
+1

**Comment by CodeNinj4** [comment]
+1

**Comment by Zhuinden** [comment]
@cmelchior interestingly, I've heard that with "stable ID support", sets would become trivial to implement in core.

However, I have this wild guess that non-sync java realm does not have stable IDs?

**Comment by ViperAlpha** [comment]
Hoping for this feature. Don't know why this is not already available...

**Comment by YuriDenison** [comment]
+1

**Comment by mlostekk** [comment]
+1

**Comment by Kelvin-D** [comment]
+1

**Comment by thehrlein** [comment]
+1

**Comment by acyuta** [comment]
+1

**Comment by leomrl** [comment]
+1

**Comment by eloraju** [comment]
+1

**Comment by sench93** [comment]
+1

**Comment by CsabaMiomni** [comment]
+2

**Comment by DoubleXStudios** [comment]
+9000

**Comment by mwshubham** [comment]
waiting for realm set :)

**Comment by tfabio** [comment]
+1

**Comment by i-schuetz** [comment]
+1 set (ordered!)

**Comment by muhrifqii** [comment]
+1

**Comment by iulian0512** [comment]
+1

**Comment by fadzlitown** [comment]
+1

**Comment by rohitgupta-kiwi** [comment]
+1

**Comment by gonharry** [comment]
+1

**Comment by BoukhariAyoub** [comment]
+1

**Comment by meSmashsta** [comment]
+1

**Comment by cmelchior** [comment]
`RealmSet` and `RealmDictionary` (Maps with String keys) was released in 10.6.0: https://github.com/realm/realm-java/blob/master/CHANGELOG.md#1060-2021-06-15.
